virtool/users/db.py

```diff
@@ -336,3 +336,27 @@ async def fetch_complete_user(mongo, user_id: str) -> Optional[User]:
         )
 
     return None
+
+
+def lookup_user_by_id(local_field: str, set_as: str = "user") -> list[Dict]:
+    return [
+        {
+            "$lookup": {
+                "from": "users",
+                "let": {"user_id": f"${local_field}"},
+                "pipeline": [
+                    {"$match": {"$expr": {"$eq": ["$_id", "$$user_id"]}}},
```

Comment:
Double check, but this can probably be simplified to something along the lines of:
```suggestion
                    {"$match": { "_id", "$$user_id"}},
```